sw/host/opentitanlib/src/app/conf.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Schema for configuration files, exact encoding json/xml to be worked out.
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum PinMode {
+    Input,
+    PushPull,
+    OpenDrain,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct PinConfiguration {
+    pub name: String,
+    pub mode: Option<PinMode>,
+    pub level: Option<bool>,
+    pub pullup: Option<bool>,
```

Comment:
Should this be an `enum PullMode`? We could then include `None`, `PullUp`, `PullDown` (and maybe weak and strong versions if we can support that on HyperDebug).